theme/scripts/patronSlider.ts

```diff
@@ -0,0 +1,161 @@
+        description:
+          "You'll get access to our community Discord, Arena, Time Exchange, patron-only events, and our archived talks earlier than the public. You'll also help sustain our programming at Index.",
+        name: "Sustaining Patron+",
+      },
+    },
+
+    SLIDER: null,
+
+    init() {
+      // Initialize noUiSlider instance.
+      PatronSlider.SLIDER = rangeSlider.create(
+        document.querySelector(PatronSlider.SELECTORS.slider),
+        {
+          // Start with the slider at this value.
+          start: [PatronSlider.START_VALUE],
+          // Add this many notches along the slider.
+          step: Object.keys(PatronSlider.PATRON_TIERS).length,
+          range: {
+            min: 5,
+            max: 25,
+          },
+          // Options for notches along the slider, called "pips"
+          pips: {
+            mode: rangeSlider.PipsMode.Steps,
+            filter: (value) => {
+              // If a slider value is not in our list of values,
+              // do not render a pip for that value.
+              if ([5, 10, 15, 20, 25].indexOf(value) > -1) {
+                return 1;
+              }
+
+              // Render a pip.
+              return -1;
+            },
+          },
+        }
+      );
+
+      const handle = document.querySelector(PatronSlider.SELECTORS.handle);
+      const patronTierLink = document.querySelector(
+        PatronSlider.SELECTORS.patronTierLink
+      );
+      const tierCopywriting = document.querySelector(
+        PatronSlider.SELECTORS.patronTierDescription
+      );
+      const patronTierName = document.querySelector(
+        PatronSlider.SELECTORS.patronTierName
+      );
+
+      // Used to set the content of our slider handle
+      handle.setAttribute("data-before", `$${PatronSlider.START_VALUE}`);
+      // Used to set the content of our drag handle.
+      handle.setAttribute("data-after", PatronSlider.MONTHLY);
+
+      patronTierLink.setAttribute(
+        "href",
+        PatronSlider.PATRON_TIERS[PatronSlider.START_VALUE].href
+      );
+
+      tierCopywriting.textContent =
+        PatronSlider.PATRON_TIERS[PatronSlider.START_VALUE].description;
+
+      patronTierName.textContent =
+        PatronSlider.PATRON_TIERS[PatronSlider.START_VALUE].name;
+
+      PatronSlider.bindEventListeners();
+    },
+
+    bindEventListeners() {
+      const pips = document.querySelectorAll(PatronSlider.SELECTORS.pips);
+      const handle = document.querySelector(PatronSlider.SELECTORS.handle);
+      const patronTierLink = document.querySelector(
+        PatronSlider.SELECTORS.patronTierLink
+      );
+      const tierCopywriting = document.querySelector(
+        PatronSlider.SELECTORS.patronTierDescription
+      );
+      const patronTierName = document.querySelector(
+        PatronSlider.SELECTORS.patronTierName
+      );
+
+      // Handle clicks on "pips", or notches, on the slider.
+      const handlePipClick = (e) => {
```

Comment:
We could make this a method on the topLevel object instead of within this function, since it takes the event as an arg.